src/main/java/gov/nasa/jpl/mbee/mdk/settings/ProjectSettings.java

```diff
@@ -0,0 +1,141 @@
+package gov.nasa.jpl.mbee.mdk.settings;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.nomagic.magicdraw.core.Application;
+import com.nomagic.magicdraw.core.Project;
+import com.nomagic.magicdraw.core.ProjectUtilities;
+import com.nomagic.uml2.ext.jmi.helpers.StereotypesHelper;
+import com.nomagic.uml2.ext.magicdraw.classes.mdkernel.Comment;
+import com.nomagic.uml2.ext.magicdraw.classes.mdkernel.Element;
+import com.nomagic.uml2.ext.magicdraw.classes.mdkernel.Package;
+import gov.nasa.jpl.mbee.mdk.json.JacksonUtils;
+import gov.nasa.jpl.mbee.mdk.util.Utils;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.swing.*;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+public class ProjectSettings {
+
+    private static Boolean warnOnce = false;
+
+    private static final Cache<Project, String> PROFILE_SERVER_CACHE = CacheBuilder.newBuilder().weakKeys().maximumSize(100).expireAfterAccess(10, TimeUnit.MINUTES).build();
+
+    public static final String
+            MMS_HOST_URL = "mms.hostUrl",
+            MMS_BASE_PATH = "mms.basePath",
+            VE_HOST_URL = "ve.hostUrl",
+            VE_BASE_PATH = "ve.basePath";
+
+
+    public static ObjectNode getProjectSettings(Project project) {
+        Package model = project.getPrimaryModel();
+        Comment comment = new ArrayList<>(model.getOwnedComment()).get(0);
+
+        String s = StringUtils.substring(comment.getBody(),StringUtils.indexOf(comment.getBody(),"{"),StringUtils.lastIndexOf(comment.getBody(),"}")+1);
```

Comment:
This is error prone, e.g. a documentation of `This model is {my favorite}` will trip it up. I would recommend using a value that must be exclusively JSON, without substring-ing. You could overload the already existing `MMS URL` tag value of the `ModelManagementSystem` stereotype by first trying to parse it as a JSON Object, then falling back to just a String to maintain backwards compatibility. This would still achieve the desired (I believe) effect of not requiring a profile migration.

Reply:
This has been the subject of nonstop discussions at this point. Adding to mms url isn’t very clear or backwards compatible. However if a user wants to indicate their model is their “{favorite}” they can choose to use the backward compatibility for the mms url only.

Reply:
I'm just seeing it for the first time and haven't been involved in the discussions. Just offering my review since it was asked for. :)

Reply:
Based on feedback; And due to it's error prone nature in testing, this has been removed in favor of built-in Project Options functionality